joomla/joomla-cms
# Modal com_modules/com_menus: improve Edit Module (+ fix old issues)

Improve modal to Add and Edit a module. (+ patch PR 10522)
**Note: to be testing on staging.**
#### Summary of Changes
- Control if modal, in edit layout
- Add <code>Save</code> (keep modal open), <code>Save & Close</code> and <code>Close</code> button to the footer of the modal
- Fix closing of Modal when edit validation returns an error (does not close the modal, and display alert message in modal)
- Fix (3.5.1) if in edit module, title is empty or changed, and then you close the modal without saving, the title of this module in the list is the same as the not saved field value in edit modal, and need a refresh of the page to be able to view it correctly back again.
- **EDIT**: Add improvement for modals rendering in Menus > Manage : viewport dimensions, close control, add Save button, add style for "Add a module to this menu" (and remove the "." at end of language string)
#### Testing Instructions (Staging)
- Go to <code>Menus</code> > <code>Add new menu item</code> > <code>Module Assignment</code> tab.
- Click on any module title to open the module edition in a modal
- Click on Save and Save & Close with a correct edition (no error), and then with an empty title, or any other possible error (not filled required field).
- Expected behavior: If an error, Save & Close won't close the modal, and Save will never close the modal (close only when requested, and on successful edition)
- Edit a module in modal, change title, and save and close : the title in the input field should be updated with the new title.
- Remove title of the module (in edit modal), to get an empty field, or edit it, and then close the modal : The title in list should be the one saved in data (and not the value not saved as in 3.5.1)

**EDIT**:
Additional tests after update : 
- Go to <code>Menus</code> > <code>Manage</code>
- Add a new menu
- Then, click on button "Add a module to this menu"
- It would now open a modal (was not opening a modal before)
- Set the module and save
- Then, you have the dropdown for modules
- Click on the module created to open the edit modal
- Do same testing as Edit module in Menu item edition (save, save&close, try with empty title, etc...)

Same rendering as for Categories in PR #10441 (screenshots)
